x-pack/plugins/ingest_manager/public/applications/ingest_manager/hooks/use_intra_app_state.tsx

```diff
@@ -57,7 +57,11 @@ export function useIntraAppState<S = AnyIntraAppRouteState>():
     // once so that it does not impact navigation to the page from within the
     // ingest app. side affect is that the browser back button would not work
     // consistently either.
-    if (location.pathname === intraAppState.forRoute && !wasHandled.has(intraAppState)) {
+
+    // Compare the routes only, strip any query params
+    const forBaseRoute = intraAppState.forRoute.split('?')[0];
```

Comment:
maybe move the parsing of the hash to line 31 since it would mean it only gets parsed once